service/src/main/java/tds/assessment/services/impl/AssessmentInfoServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package tds.assessment.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import tds.assessment.AssessmentInfo;
+import tds.assessment.AssessmentWindow;
+import tds.assessment.repositories.AssessmentQueryRepository;
+import tds.assessment.services.AssessmentInfoService;
+import tds.assessment.services.AssessmentWindowService;
+
+@Service
+public class AssessmentInfoServiceImpl implements AssessmentInfoService {
+    private final AssessmentQueryRepository assessmentQueryRepository;
+    private final AssessmentWindowService assessmentWindowService;
+
+    @Autowired
+    public AssessmentInfoServiceImpl(final AssessmentQueryRepository assessmentQueryRepository,
+                                     final AssessmentWindowService assessmentWindowService) {
+        this.assessmentQueryRepository = assessmentQueryRepository;
+        this.assessmentWindowService = assessmentWindowService;
+    }
+
+    @Override
+    public List<AssessmentInfo> findAssessmentInfo(final String clientName, final String... assessmentKeys) {
+        List<AssessmentInfo> assessments = assessmentQueryRepository.findAssessmentInfoByKeys(clientName, assessmentKeys);
+        Set<String> assessmentIds = assessments.stream().map(AssessmentInfo::getId).collect(Collectors.toSet());
+        Map<String, List<AssessmentWindow>> assessmentWindows = assessmentWindowService.findAssessmentWindowsForAssessmentIds(
+            clientName, assessmentIds.toArray(new String[assessmentIds.size()]));
```

Comment:
I have to admit, this feels a little funky.  If it's going to be called via code, I'd recommend switching the findAssessmentWindowsForAssessmentIds method to accept a collection rather than a varargs param.